File: MicroBenchmarks/For-Loop-vs-Lexical-Scope-Stress.js
var N = 10000000;

var acc = 0;

function f(x) { return acc + (10 * x); }

var acc, t0, elapsed;
WScript.StdOut.WriteLine("MS JScript... About to stress the for loop vs lexical scope " + N.toString() + " times...");
WScript.StdOut.WriteLine("(Press return)");
WScript.StdIn.ReadLine();

t0 = new Date().valueOf();
for(var i = 0; i < N; i++) acc = f(i);
elapsed = new Date().valueOf() - t0;

WScript.StdOut.WriteLine("acc = " + acc.toString() + " in " + elapsed.toString() + " ms");
WScript.StdOut.WriteLine("(Press return to exit)");
WScript.StdIn.ReadLine();

File: MicroBenchmarks/For-Loop-vs-Lexical-Scope-Stress.py
from time import time

def started_time():
	return int(1000 * time())

def elapsed_time(start):
	return int(1000 * time()) - start

N = 10000000

acc = 0

def f(x):
	return acc + (10 * x)

print("Python... About to stress the for loop vs lexical scope " + str(N) + " times...")
print("(Press return)")
input()

t0 = started_time()
for i in range(0, N):
	acc = f(i)
elapsed = elapsed_time(t0)
print("acc = " + str(acc) + " in " + str(elapsed) + " ms")
print("(Press return to exit)")
input()
